cmd/swarm/clientaccountmanager.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2016 The go-ethereum Authors
+// This file is part of go-ethereum.
+//
+// go-ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// go-ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+
+/*
+//This module implements the bare minimum to allow client commands to access the keystore
+*/
+package main
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/accounts/usbwallet"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+	cli "gopkg.in/urfave/cli.v1"
+
+	"github.com/ethereum/go-ethereum/cmd/utils"
+)
+
+type ClientAccountManagerProvider struct {
+	manager *accounts.Manager
+}
+
+func (c *ClientAccountManagerProvider) AccountManager() *accounts.Manager {
+	return c.manager
+}
+
+func getClientAccount(ctx *cli.Context) *ecdsa.PrivateKey {
+	bzzconfig, err := buildConfig(ctx)
+	if err != nil {
+		utils.Fatalf("Cannot read configuration: %s", err.Error())
+		return nil
+	}
+	nodeconfig := defaultNodeConfig
+	if _, err := os.Stat(bzzconfig.Path); err == nil {
+		nodeconfig.DataDir = bzzconfig.Path
+	}
+
+	accountManager, _, err := makeAccountManager(&nodeconfig)
+	if err != nil {
+		utils.Fatalf("Cannot create account manager: %s", err.Error())
+		return nil
+	}
+
+	return getAccount(bzzconfig.BzzAccount, ctx, &ClientAccountManagerProvider{
+		manager: accountManager,
+	})
+
+}
+
+func makeAccountManager(conf *node.Config) (*accounts.Manager, string, error) {
+	scryptN, scryptP, keydir, err := conf.AccountConfig()
+	var ephemeral string
+	if keydir == "" {
+		// There is no datadir.
+		keydir, err = ioutil.TempDir("", "go-ethereum-keystore")
```

Comment:
I still don't understand this temporary directory stuff here. Did you do any looking into why this is solved in this manner? Where did the code come from again? (I think you said you cut and pasted it from somewhere?)

Reply:
This comes from `/node/config.go`

Reply:
well, at the very least the key should come from our actual keystore, andr _fail_ if there is no key (or keystore) there. To create a new key, users should use `geth account new`

Reply:
I digged a bit deeper into the ephemeral directory stuff. An Ethereum node will create an ephemeral keystore folder that will be deleted once the node is shutdown, if a folder cannot be set up. This comes from the following:
```
// KeyStoreDir is the file system folder that contains private keys. The directory can
	// be specified as a relative path, in which case it is resolved relative to the
	// current directory.
	//
	// If KeyStoreDir is empty, the default location is the "keystore" subdirectory of
	// DataDir. If DataDir is unspecified and KeyStoreDir is empty, an ephemeral directory
	// is created by New and destroyed when the node is stopped.
```
In any case, I have disabled ephemeral directory creation. If a keystore is not present, it will fail with an error.